From 32567a2d87cd39ae48b4c05bc470633f9dd741eb Mon Sep 17 00:00:00 2001
From: Qiang Yu <yuq825@gmail.com>
Date: Wed, 21 Jun 2017 14:45:38 +0800
Subject: [PATCH 066/134] drm/lima: move vm free to drm postclose callback

this makes vm free after gem bo release.
otherwise the gem bo unmap will use a vm which has
already been freed and cause kernel error.

Signed-off-by: Qiang Yu <yuq825@gmail.com>
---
 drivers/gpu/drm/lima/lima_drv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/lima/lima_drv.c b/drivers/gpu/drm/lima/lima_drv.c
index b8625fa..b1ed44b 100644
--- a/drivers/gpu/drm/lima/lima_drv.c
+++ b/drivers/gpu/drm/lima/lima_drv.c
@@ -204,7 +204,7 @@ static int lima_drm_driver_open(struct drm_device *dev, struct drm_file *file)
 	return err;
 }
 
-static void lima_drm_driver_preclose(struct drm_device *dev, struct drm_file *file)
+static void lima_drm_driver_postclose(struct drm_device *dev, struct drm_file *file)
 {
 	struct lima_drm_priv *priv = file->driver_priv;
 
@@ -240,7 +240,7 @@ static struct drm_driver lima_drm_driver = {
 	.load		    = lima_drm_driver_load,
 	.unload             = lima_drm_driver_unload,
 	.open               = lima_drm_driver_open,
-	.preclose           = lima_drm_driver_preclose,
+	.postclose          = lima_drm_driver_postclose,
 	.ioctls             = lima_drm_driver_ioctls,
 	.num_ioctls         = ARRAY_SIZE(lima_drm_driver_ioctls),
 	.fops               = &lima_drm_driver_fops,
-- 
2.0.1

